xidel: add page (#7678)
tldr-pages/tldr: pages/common/xidel.md

# xidel

> > A command line tool to download and extract data from HTML/XML pages as well as JSON APIs.

Reviewer: ```suggestion
>Download and extract data from HTML/XML pages as well as JSON APIs.
```

> More information: <https://www.videlibri.de/xidel.html>.

> - Print all URLs found by a google search:

Reviewer 2: ```suggestion
- Print all URLs found by a Google search:
```

> - Print all URLs found by a google search:

Reviewer 3: I feel like a number of the examples on this page are too specific to a single website. Can we generalise them at all?

Reviewer 4: I'd like to add, the google links are .de, rather than a default google.com

Author: Not sure how to work with first comment.  I took the examples straight from the main website.  3-5 seem to be the best basic ones to use.

> `xidel https://www.google.com/search?q=test --extract "//a/extract(@href, 'url[?]q=([^&]+)&', 1)[. != '']"`

Reviewer 2: ```suggestion
`xidel {{https://www.google.com/search?q=test}} --extract "//a/extract(@href, 'url[?]q=([^&]+)&', 1)[. != '']"`
```

> - Print the title of all pages found by a google search and download them:

Reviewer 2: ```suggestion
- Print the title of all pages found by a Google search and download them:
```

> `xidel https://www.google.de/search?q=test --follow "//a/extract(@href, 'url[?]q=([^&]+)&', 1)[. != '']" --extract //title --download '{$host}/'`

Reviewer 5: ```suggestion
`xidel {{https://www.google.de/search?q=test}} --follow "{{//a/extract(@href, 'url[?]q=([^&]+)&', 1)[. != '']}}" --extract {{//title}} --download {{'{$host}/'}}`
```

> - Follow all links on a page and print the titles, With XPath:

Reviewer 4: ```suggestion
- Follow all links on a page and print the titles, with XPath:
```

Could you please change this throughout

> `xidel https://example.org -f //a -e //title`

Reviewer 5: ```suggestion
`xidel {{https://example.org}} -f {{//a}} -e {{//title}}`
```

> `xidel {{https://example.org}} -f {{//a}} -e {{//title}}`

Reviewer 2: Use long arguments
```suggestion
`xidel {{https://example.org}} --follow {{//a}} --extract {{//title}}`
```

- Follow all links on a page and print the titles, With CSS selectors:

> `xidel https://example.org -f "css('a')" --css title`

Reviewer 5: ```suggestion
`xidel {{https://example.org}} -f "{{css('a')}}" --css {{title}}`
```

> `xidel {{https://example.org}} -f "{{css('a')}}" --css {{title}}`

Reviewer 2: ```suggestion
`xidel {{https://example.org}} --follow "{{css('a')}}" --css {{title}}`
```

- Follow all links on a page and print the titles, With pattern matching:

> `xidel https://example.org -f "<a>{.}</a>*" -e "<title>{.}</title>"`

Reviewer 5: ```suggestion
`xidel {{https://example.org}} -f "{{<a>{.}</a>*}}" -e "{{<title>{.}</title>}}"`
```

> `xidel {{https://example.org}} -f "{{<a>{.}</a>*}}" -e "{{<title>{.}</title>}}"`

Reviewer 2: ```suggestion
`xidel {{https://example.org}} --follow "{{<a>{.}</a>*}}" --extract "{{<title>{.}</title>}}"`
```

- If you have an example.xml file, you can read the important part like (which will also check, if the element containing "ood" is there, and fail otherwise):

> `xidel example.xml -e "<x><foo>ood</foo><bar>{.}</bar></x>"`

Reviewer 5: ```suggestion
`xidel {{example.xml}} -e "{{<x><foo>ood</foo><bar>{.}</bar></x>}}"`
```

> `xidel {{path/to/example.xml}} -e "{{<x><foo>ood</foo><bar>{.}</bar></x>}}"`

Reviewer 2: ```suggestion
`xidel {{path/to/example.xml}} --extract "{{<x><foo>ood</foo><bar>{.}</bar></x>}}"`
```

> - If you have an example.xml file, you can read the important part like (which will also check, if the element containing "ood" is there, and fail otherwise):
> 
> `xidel {{example.xml}} -e "{{<x><foo>ood</foo><bar>{.}</bar></x>}}"`

Reviewer: ```suggestion
- Read the pattern from example.xml (which will also check if the element containing "ood" is there, and fail otherwise):

`xidel {{path/to/example.xml}} -e "{{<x><foo>ood</foo><bar>{.}</bar></x>}}"`
```

> - Print all newest Stackoverflow questions with title and url using pattern matching on their RSS feed:

Reviewer 2: ```suggestion
- Print all newest Stack Overflow questions with title and URL using pattern matching on their RSS feed:
```

> `xidel http://stackoverflow.com/feeds -e "<entry><title>{title:=.}</title><link>{uri:=@href}</link></entry>+"`

Reviewer 5: ```suggestion
`xidel {{http://stackoverflow.com/feeds}} -e "{{<entry><title>{title:=.}</title><link>{uri:=@href}</link></entry>+}}"`
```

> `xidel {{http://stackoverflow.com/feeds}} -e "{{<entry><title>{title:=.}</title><link>{uri:=@href}</link></entry>+}}"`

Reviewer 2: ```suggestion
`xidel {{http://stackoverflow.com/feeds}} --extract "{{<entry><title>{title:=.}</title><link>{uri:=@href}</link></entry>+}}"`
```

- Check if you have Reddit mail, Webscraping, combining CSS, XPath, JSONiq, and automatically form evaluation:

> `xidel https://reddit.com -f "form(css('form.login-form')[1], {'user': '$your_username', 'passwd': '$your_password'})" -e "css('#mail')/@title"`

Reviewer 5: ```suggestion
`xidel {{https://reddit.com}} -f "{{form(css('form.login-form')[1], {'user': '$your_username', 'passwd': '$your_password'})}}" -e "{{css('#mail')/@title}}"`
```

> `xidel {{https://reddit.com}} -f "{{form(css('form.login-form')[1], {'user': '$your_username', 'passwd': '$your_password'})}}" -e "{{css('#mail')/@title}}"`

Reviewer 2: ```suggestion
`xidel {{https://reddit.com}} --follow "{{form(css('form.login-form')[1], {'user': '$your_username', 'passwd': '$your_password'})}}" --extract "{{css('#mail')/@title}}"`
```
